[Docs] `Touchable` (#4022)
kmagiera/react-native-gesture-handler: packages/docs-gesture-handler/docs/guides/upgrading-to-3.mdx

### Buttons

> RNGH3 introduces the [`Touchable`](/docs/components/touchable) component — a flexible, unified replacement for all previous button types. While `Touchable` shares the same logic as our standard buttons, it offers a more customizable API.

Reviewer: ```suggestion
RNGH3 introduces the [`Touchable`](/docs/components/touchable) component — a flexible, unified replacement for all previous button types. While `Touchable` shares most of the logic with our standard buttons, it offers a more customizable API.
```

To help you migrate, here is the current state of our button components:

- [`Touchable`](/docs/components/touchable) - The recommended component. Can be used to replicate both `RectButton` and `BorderlessButton` effects by adjusting the `activeOpacity` and `activeUnderlayOpacity` props.
  - To replace `RectButton`, add `activeUnderlayOpacity={0.105}`.

>   - To replace `BorderlessButton`, add `activeOpacity={0.3}`.

Reviewer: Do we want to also list legacy touchables here?

Author: I think we can. Touchables were deprecated long time ago, but we can also show how to replace them. We've already told users to use `Pressable`, but maybe `Touchable` will fit better for some of them. 

However, I don't think it fits in `Buttons` section. I've added separate one for `Touchables` in [155f891](https://github.com/software-mansion/react-native-gesture-handler/pull/4022/commits/155f8919b5a6d4e8ddae587aadcdaa88305975cc), let me know what you think

Reviewer: Looks good!
